packages/riverpod/lib/src/framework/value_provider.dart

```diff
@@ -56,7 +56,7 @@ class ValueProviderElement<State> extends ProviderElementBase<State> {
         }(),
         '',
       );
-      setState(newValue);
+      _markDependencyChanged();
```

Comment:
setState should call _markDependencyChanged anyway

Reply:
That's breaking stuff though, as you can see in the CI.

There's probably a different solution.

Reply:
The TODO app that I see that breaks in tests works fine when I run it, but the test fails.  Interesting.

Reply:
Your change is bound to be problematic.

I can guarantee that this is not the fix. ValueProvider isn't the problem here